contracts/peripheral/PoolProxy.sol

```diff
@@ -0,0 +1,346 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.10;
+
+import "../interfaces/IWeth.sol";
+import "../interfaces/IDai.sol";
+import "../interfaces/IGemJoin.sol";
+import "../interfaces/IDaiJoin.sol";
+import "../interfaces/IVat.sol";
+import "../interfaces/IPot.sol";
+import "../interfaces/IPool.sol";
+import "../interfaces/IFYDai.sol";
+import "../interfaces/IChai.sol";
+import "../interfaces/IDelegable.sol";
+import "../interfaces/ITreasury.sol";
+import "../interfaces/IController.sol";
+import "../helpers/DecimalMath.sol";
+
+
+library SafeCast {
+    /// @dev Safe casting from uint256 to uint128
+    function toUint128(uint256 x) internal pure returns(uint128) {
+        require(
+            x <= type(uint128).max,
+            "YieldProxy: Cast overflow"
+        );
+        return uint128(x);
+    }
+
+    /// @dev Safe casting from uint256 to int256
+    function toInt256(uint256 x) internal pure returns(int256) {
+        require(
+            x <= uint256(type(int256).max),
+            "YieldProxy: Cast overflow"
+        );
+        return int256(x);
+    }
+}
+
+contract PoolProxy is DecimalMath {
+    using SafeCast for uint256;
+
+    IVat public vat;
+    IWeth public weth;
+    IDai public dai;
+    IGemJoin public wethJoin;
+    IDaiJoin public daiJoin;
+    IChai public chai;
+    IController public controller;
+    ITreasury public treasury;
+
+    IPool[] public pools;
+    mapping (address => bool) public poolsMap;
+
+    bytes32 public constant CHAI = "CHAI";
+    bytes32 public constant WETH = "ETH-A";
+    bool constant public MTY = true;
+    bool constant public YTM = false;
+
+
+    constructor(address controller_, IPool[] memory _pools) public {
+        controller = IController(controller_);
+        treasury = controller.treasury();
+
+        weth = treasury.weth();
+        dai = IDai(address(treasury.dai()));
+        chai = treasury.chai();
+        daiJoin = treasury.daiJoin();
+        wethJoin = treasury.wethJoin();
+        vat = treasury.vat();
+
+        // for repaying debt
+        dai.approve(address(treasury), uint(-1));
+
+        // for posting to the controller
+        chai.approve(address(treasury), uint(-1));
+        weth.approve(address(treasury), uint(-1));
+
+        // for converting DAI to CHAI
+        dai.approve(address(chai), uint(-1));
+
+        vat.hope(address(daiJoin));
+        vat.hope(address(wethJoin));
```

Comment:
Are these needed? I think we only wanted them for the Maker integration

Reply:
Not needed. Removed a bunch of other unneeded stuff as well.